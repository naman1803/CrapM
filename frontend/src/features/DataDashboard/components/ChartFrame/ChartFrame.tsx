import {
  Box,
  Grid2,
  SxProps,
} from '@mui/material';
import React from 'react';

// union for available dashboard sizings
type DashboardComponentSizing = 'row' | 'half' | 'quarter' | 'box';

export interface ChartFrameProps {
  size: DashboardComponentSizing;
  containerStyles?: SxProps;
  children: React.ReactNode;
}

/**
 * General Chart Component Frame
 */
const ChartFrame: React.FC<ChartFrameProps> = ({ size, containerStyles, children }) => {

  // dict for available grid sizings
  const gridSizings = {
    row: { xs: 12 },
    half: { xs: 12, md: 6 },
    quarter: { xs: 12, md: 6, lg: 3 },
    box: { xs: 12, sm: 6, md: 4, lg: 3 },
  };

  return (
    <Grid2 size={gridSizings[size]}>
      <Box
        sx={{
          border: 1,
          borderColor: '#3a3a3a',
          borderRadius: 1,
          ...containerStyles,
        }}
        padding={'0.5rem'}
      >
        <Grid2
          size={12}
          padding={'0.5rem'}
        >
        </Grid2>
        <Grid2 size={12} height={'100%'}>
          {children}
        </Grid2>
      </Box>
    </Grid2>
  );
};
export default ChartFrame;
