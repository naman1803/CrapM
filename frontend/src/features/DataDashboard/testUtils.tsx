jest.mock('recharts', () => ({
  __esModule: true,
  ...jest.requireActual('recharts'),
  ResponsiveContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>ResponsiveContainer</p>
      {children}
    </div>
  )),
}));
