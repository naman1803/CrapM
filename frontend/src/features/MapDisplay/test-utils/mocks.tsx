import {
  LakeGeoPoint,
  MapElement,
  MapElementType,
  SewageGeoPoint,
} from '../types/MapElements';

export const mockLakeElements: MapElement[] = [
  {
    id: 1,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      name: 'Grrrr Lake',
      additional_notes: 'This is a lake',
    } as LakeGeoPoint,
  },
  {
    id: 2,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      name: 'Obama Lake',
    } as LakeGeoPoint,
  },
];

export const mockSewageElements: MapElement[] = [
  {
    id: 1,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      facility_owner: 'Eric',
      radius: 100,
      colour: 'Blue',
    } as SewageGeoPoint,
  },
  {
    id: 2,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      facility_owner: 'Myron',
      radius: 100,
      colour: 'Red',
    } as SewageGeoPoint,
  },
];
