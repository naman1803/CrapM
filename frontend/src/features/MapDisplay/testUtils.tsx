import { LeafletEventHandlerFnMap, LatLngExpression } from 'leaflet';

jest.mock('react-leaflet', () => ({
  __esModule: true,
  ...jest.requireActual('leaflet'),
  useMap: () => ({
    flyTo: jest.fn(),
    getZoom: jest.fn().mockReturnValue(13), // Add getZoom method returning default zoom level
    on: jest.fn(), // Mock on method for event handling
    off: jest.fn(), // Mock off method for removing event listeners
  }),
  MapContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Map Container</p>
      {children}
    </div>
  )),
  Marker: jest
    .fn()
    .mockImplementation(
      ({
        position,
        children,
        key,
        eventHandlers,
      }: {
        eventHandlers: LeafletEventHandlerFnMap;
        position: LatLngExpression;
        key: any;
        children: React.ReactNode;
      }) => (
        <div key={key}>
          <p>{`Marker-${position.toString()}`}</p>
          {Object.entries(eventHandlers).map(([keyE, value]) => (
            <button key={keyE} onClick={value}>
              {`button-${keyE}-${position.toString()}`}
            </button>
          ))}
          {children}
        </div>
      ),
    ),
  Popup: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Popup</p>
      {children}
    </div>
  )),
  TileLayer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>TileLayer</p>
      {children}
    </div>
  )),
}));
