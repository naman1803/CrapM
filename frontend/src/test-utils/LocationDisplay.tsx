import { useLocation } from 'react-router-dom';
/**test component which displays the current url location.
 * use testid = location-display to get value of current location
 */
const LocationDisplay: React.FC = () => {
  const location = useLocation();

  return <div data-testid='location-display'>{location.pathname}</div>;
};
export default LocationDisplay;
