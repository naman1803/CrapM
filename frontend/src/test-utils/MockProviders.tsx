import { ThemeProvider } from '@mui/material';
import { MemoryRouter } from 'react-router-dom';
import { themeOptions } from '@app/theme';
import React from 'react';

// Wraps all providers for the app for testing
export const TestProviders = ({
  children,
  initialRoute = '/',
}: {
  children?: React.ReactNode;
  initialRoute?: string;
}) => {
  return (
    <ThemeProvider theme={themeOptions}>
      <MemoryRouter initialEntries={[initialRoute]}>{children}</MemoryRouter>
    </ThemeProvider>
  );
};
