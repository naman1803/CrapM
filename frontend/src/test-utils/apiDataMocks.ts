import { LakeData } from '@/types/LakeData';
import {
  SewageCoordData,
  SewageData,
  SewageSeverityData,
} from '@/types/SewageData';

export const mockSewageSeverityElement: SewageSeverityData = {
  id: 1,
  facilityOwner: 'Eric',
  contaminants: {
    brainrot: {
      averageSeverityScore: 1.0,
      overallSeverityLevel: 'Moderate',
      averagePercentExceedance: 100.0,
      records: [
        {
          severityScore: 1.0,
          severityLevel: 'Moderate',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
    takingDumb: {
      averageSeverityScore: 2.0,
      overallSeverityLevel: 'High',
      averagePercentExceedance: 200.0,
      records: [
        {
          severityScore: 0.5,
          severityLevel: 'High',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
  },
};

export const mockLakeApiData: LakeData[] = [
  {
    id: 1,
    lakeName: 'Lake Ontario',
    latitude: 45.2,
    longitude: -78.94,
    additionalNotes: null,
  },
  {
    id: 2,
    lakeName: 'Lake Erie',
    latitude: 46.47,
    longitude: -82.78,
    additionalNotes: null,
  },
];

export const mockSewageApiDataSent = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    contaminant: 'PH - HIGH',
    radius: 22.524761,
    colour: "Green"
  },
];
export const mockSewageApiData: SewageCoordData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    radius: 22.524761,
    colour: "Green"
  },
];

export const mockSewageSeverityApiData: SewageSeverityData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    contaminants: {
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.77,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 177.0,
        records: [
          {
            severityScore: 0.77,
            severityLevel: 'Low',
            date: '2016-08-18',
          },
        ],
      },
    },
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    contaminants: {
      'PH - HIGH': {
        averageSeverityScore: 0.95,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 195.4,
        records: [
          {
            severityScore: 0.95,
            severityLevel: 'Low',
            date: '2016-08-30',
          },
        ],
      },
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.58,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 157.8,
        records: [
          {
            severityScore: 0.58,
            severityLevel: 'Low',
            date: '2016-08-23',
          },
        ],
      },
    },
  },
];

export const mockSewageAllApiData: SewageData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    siteAddress: '123 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-18T00:00:00',
    exceedanceEndDate: '2016-08-18T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 88.5,
    quantityMinimum: null,
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-23T00:00:00',
    exceedanceEndDate: '2016-08-23T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 78.9,
    quantityMinimum: null,
  },
  {
    id: 3,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'PH - HIGH',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-30T00:00:00',
    exceedanceEndDate: '2016-08-30T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 97.7,
    quantityMinimum: null,
  },
];
