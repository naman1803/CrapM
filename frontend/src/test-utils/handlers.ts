import { http, HttpResponse } from 'msw';
import {
  mockLakeApiData,
  mockSewageAllApiData,
  mockSewageApiDataSent,
  mockSewageSeverityApiData,
} from './apiDataMocks';

const handlers = [
  http.get('/api/maps/lakes/all', () => {
    return HttpResponse.json(mockLakeApiData);
  }),
  http.get('/api/maps/sewage/all', () => {
    return HttpResponse.json(mockSewageApiDataSent);
  }),
  http.get('/api/data/sewage/all', () => {
    return HttpResponse.json(mockSewageAllApiData);
  }),
  ...mockSewageSeverityApiData.map((severityData) =>
    http.get(`/api/data/sewage/severity/${severityData.id}`, () => {
      return HttpResponse.json(severityData);
    }),
  ),
];
export default handlers;
