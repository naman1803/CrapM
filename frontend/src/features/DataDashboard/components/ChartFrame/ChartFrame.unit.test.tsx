import './../../testUtils';

import { TestProviders } from "@/test-utils/MockProviders";
import { render } from "@testing-library/react";
import ChartFrame, { ChartFrameProps } from "./ChartFrame";
import { Typography } from "@mui/material";

const mockChartFrame: ChartFrameProps = {
    size: 'half',
    children: <Typography>Hello World</Typography>
};

describe('ChartFrame', () => {
    it('should render the chart', () => {
        const { asFragment } = renderChartFrame(mockChartFrame);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should display its child components', () => {
        const { getByText } = renderChartFrame(mockChartFrame);
        expect(getByText('Hello World')).toBeInTheDocument();
    });
});

const renderChartFrame = (props: ChartFrameProps) => render(
    <TestProviders>
        <ChartFrame {...props} />
    </TestProviders>
);
