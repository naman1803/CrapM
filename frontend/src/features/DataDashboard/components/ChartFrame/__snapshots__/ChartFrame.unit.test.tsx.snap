// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartFrame should render the chart 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 css-1odx5a8-MuiGrid2-root"
  >
    <div
      class="MuiBox-root css-1pnkh5u"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-w3fsr2-MuiGrid2-root"
      />
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1e8ggq8-MuiGrid2-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-tckkrt-MuiTypography-root"
        >
          Hello World
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
